.vsts-ci/linux.yml

```diff
@@ -95,24 +100,69 @@ stages:
     parameters:
       pool: ubuntu-20.04
 
+- stage: TestContainer
+  displayName: Test in a container
+  dependsOn: [BuildLinuxStage]
+  jobs:
+  - job: getContainerJob
+    displayName: Choose a container
+    pool:
+      vmImage: ubuntu-20.04
+    steps:
+    - checkout: self
+      clean: true
+
+    - checkout: Docker
+      clean: true
+
+    - pwsh: |
+        $matrix = ./PowerShell-Docker/build.ps1 -GenerateMatrixJson -FullJson -Channel preview | ConvertFrom-Json
+        $linuxImages = $matrix.preview |
+          Where-Object {$_.osversion -notlike '*windows*' -and $_.osversion -notlike '*nano*' -and $_.osversion -notlike 'alpine*' -and $_.osversion -notlike '*arm*'} |
+          Select-Object -First 100  JobName, Taglist |
+          Sort-Object -property JobName
+        $marinerImage = $linuxImages | where-object {$_.JobName -like 'mariner*'}
+        Import-Module ./PowerShell/tools/ci.psm1
+        $tag = $marinerImage.Taglist -split ';' | select-object -first 1
+        $containerName = "mcr.microsoft.com/powershell/test-deps:$tag"
+        Set-BuildVariable -Name containerName -Value $containerName -IsOutput
+        Set-BuildVariable -Name containerBuildName -Value $marinerImage.JobName -IsOutput
+      name: getContainerTask
+      displayName: Get Container
+      continueOnError: true
+
+  - template: templates/test/nix-container-test.yml
+    parameters:
+      name: container
+      pool: ubuntu-20.04
+      purpose: UnelevatedPesterTests
+      tagSet: CI
+
+  - template: templates/test/nix-container-test.yml
+    parameters:
+      name: container
+      pool: ubuntu-20.04
+      purpose: ElevatedPesterTests
+      tagSet: CI
+
+  - template: templates/test/nix-container-test.yml
+    parameters:
+      name: container
+      pool: ubuntu-20.04
+      purpose: UnelevatedPesterTests
+      tagSet: Others
+
+  - template: templates/test/nix-container-test.yml
+    parameters:
+      name: container
+      pool: ubuntu-20.04
+      purpose: ElevatedPesterTests
+      tagSet: Others
+
 - stage: PackageLinux
   displayName: Package Linux
-  dependsOn: ["BuildLinux"]
+  dependsOn: ["BuildLinuxStage"]
   jobs:
   - template: linux/templates/packaging.yml
     parameters:
       pool: ubuntu-20.04
-
-- stage: CodeCovTestPackage
```

Comment:
On checking a part of this is required, where we create test packages.

Reply:
https://github.com/PowerShell/PowerShell/blob/36fcec3bddc2e6bab20afced9fe2451bda6c503e/tools/ci.psm1#L373

Reply:
spoke offline and we use this from another pipeline.